### DISCUSSION TITLE
TUL PYNQ Z2's Misleading Xilinx Part Name

### DISCUSSION THREAD

**Discussion by Fe-r-oz** [fastmachinelearning/hls4ml#702] [Q&A] [answered]
Hi @thesps. Hope you are doing great.

There is an issue with the Vivado accelerator backend-board-name file that I would like to bring to your kind attention.

TUL PYNQ-Z2 has a Xilinx part name of XC7Z020-1CLG400C instead of xc7z020clg400-1 as written in the backend support file for the Vivado accelerator. 

In the research paper titled "FAST Convolutional Neural Networks on FPGAs with HLS4ML", the Xilinx part is mentioned to be XC7Z020CLG400-1 but on the TUL PYNQ Z2 website, the Xilinx part name is mentioned as XC7Z020-1CLG400C.

I have also checked the exact specifications that are mentioned in this research paper as well. It matches with specifications, but the Xilinx part is XC7Z020-1CLG400C. See website https://www.xilinx.com/support/university/xup-boards/XUPPYNQ-Z2.html

This means that the PYNQ Z2 Xilinx part name that is written in the Vivado accelerator backend is misleading as it directs one to a totally different board than what the Xilinx part name suggests. 

Hence, there is skepticism because this XC7Z020-1CLG400C is not supported by HLS4ML but it's mentioned that pynq z2 board is supported by Vivado accelerator which it is, but I believe Xilinx Part name in the backend file is wrong/misleading.

![ActualXilinxPart](https://user-images.githubusercontent.com/93876775/210192776-8a1eec6b-0324-483f-b306-f8b7e9573ea5.PNG)
-0fd5a793cab6.png)

![ZYNQ-7000-Instead](https://user-images.githubusercontent.com/93876775/210193041-7743aabd-3577-4c75-86d5-35067f3d5bc6.png)

I hope more clarity would ensue via discussion. Kindly correct me in case I am wrong.

**Comment by jmduarte** [comment, marked as answer]
Hi @Fe-r-oz this is simply the convention for how the Vivado/Vitis tools specify the part number so it's out of our control. As you can see it's just a bit of transposition from what you circle.

The `get_parts` TCL command (https://docs.xilinx.com/r/en-US/ug835-vivado-tcl-commands/get_parts) is useful to get a listing of all supported part numbers
